5_analyze_data/notebooks/Heatmap_analysis/nf1_complexheatmap.r

```diff
@@ -0,0 +1,155 @@
+suppressPackageStartupMessages(library(ComplexHeatmap))
+suppressPackageStartupMessages(library(dplyr))
+
+# Set paths and constants
+input_data_dir <- file.path("..", "..", "..", "4_processing_features", "data")
+output_figure_dir <- "figures"
+
+cp_heatmap_file <- file.path(output_figure_dir, "cp_complex_heatmap.pdf")
+dp_heatmap_file <- file.path(output_figure_dir, "dp_complex_heatmap.pdf")
+
+# Set heatmap colors
+well_cols = c(
+    "C6" = "#E1DAAE",
+    "C7" = "#FF934F",
+    "D6" = "#CC2D35",
+    "D7" = "#058ED9",
+    "E6" = "#848FA2",
+    "E7" = "#2D3142",
+    "F6" = "#FFC857",
+    "F7" = "#5f7a12"
+)
+genotype_cols = c(
+    "Null" = "#785EF0",
+    "WT" = "#DC267F"
+)
+
+# Load data
+cp_file <- file.path(input_data_dir, "nf1_sc_norm_fs_cellprofiler.csv.gz")
+
+cp_df <- readr::read_csv(
+    cp_file,
+    col_types = readr::cols(
+        .default="d",
+        Metadata_WellRow="c",
+        Metadata_WellCol="c",
+        Metadata_Well="c",
+        Metadata_gene_name="c",
+        Metadata_genotype="c"
+    )
+) %>% dplyr::select(-...1)  # Drop index col
+
+print(dim(cp_df))
+head(cp_df, 3)
+
+# Split metadata and feature data
+cp_metadata_df <- cp_df %>% dplyr::select(tidyr::starts_with("Metadata"))
+cp_meta_cols <- colnames(cp_metadata_df)
+cp_df <- cp_df %>% dplyr::select(-!!cp_meta_cols)
+
+# Calculate correlation matrix from feature data
+cp_cor_matrix <- t(cp_df) %>% cor()
+
+print(dim(cp_cor_matrix))
+head(cp_cor_matrix, 3)
+
+ht <- Heatmap(
+    cp_cor_matrix,
+    name = "Pearson\nCorrelation",
+    column_dend_side = "top",
+    
+    clustering_method_columns = "average",
+    clustering_method_rows = "average",
+    
+    top_annotation = HeatmapAnnotation(
+        Genotype = cp_metadata_df$Metadata_genotype,
+        CellCount = anno_barplot(
+            cp_metadata_df$Metadata_number_of_singlecells,
+            height = unit(1, "cm")
+        ),
+        Well = cp_metadata_df$Metadata_Well,
+        
+        col = list(
+            Genotype = genotype_cols,
+            Well = well_cols
+        )
+    )
+)
+
+draw(ht)
+
+# Save heatmap to file
+pdf(cp_heatmap_file)
```

Comment:
Why save the figure as a pdf and not a png like the other figures?

Reply:
I generally like how PDFs look better than PNGs, but PNGs are really nifty on github. I also much prefer PDFs for ComplexHeatmap for some reason.

PNGs added in e8ccc7581d60efcb2a9b34165693101a592cb1c2 (I think it's good to add both, since github can deal with PNGs much better)

Thanks!